sdks/java/io/cdap/src/test/java/org/apache/beam/sdk/io/cdap/TestOutputDoFn.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.cdap;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.transforms.DoFn;
+
+public class TestOutputDoFn extends DoFn<String, String> {
+  private static final List<String> RECORDS = new ArrayList<>();
+
+  @ProcessElement
+  public void processElement(@Element String input, DoFn.OutputReceiver<String> output) {
+    RECORDS.add(input);
```

Comment:
Note that this pattern will not work in general since runners might choose to split bundles and/or may serialize the DoFn class before execution. Please use PAssert for verification instead.

Reply:
Removed